staging/src/k8s.io/api/admissionregistration/v1beta1/types.go

```diff
@@ -576,31 +592,575 @@ type ServiceReference struct {
+	//
+	// "Deny" and "Warn" may not be used together since this combination
+	// needlessly duplicates the validation failure both in the
+	// API response body and the HTTP warning headers.
 	//
 	// Required.
-	Expression string `json:"expression" protobuf:"bytes,2,opt,name=expression"`
+	// +listType=set
+	ValidationActions []ValidationAction `json:"validationActions,omitempty" protobuf:"bytes,4,rep,name=validationActions"`
+}
+
+// BindingParamRef describes how to locate the param used as input to the a
+// expression of rule applied by a policy binding.
+// +union
+type BindingParamRef struct {
+	// `kind` indicates which one of the other fields is non-empty.
+	// Required
+	// +unionDiscriminator
+	Kind BindingParamKind `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// GlobalParamRef specifies the parameter resource used to configure the
+	// admission control policy.
+	// If the resource referred to by ParamRef does not exist, this binding is
+	// considered mis-configured and the FailurePolicy of the
+	// ValidatingAdmissionPolicy applied.
+	// +optional
+	GlobalParamRef *ParamRef `json:"globalParamRef,omitempty" protobuf:"bytes,2,rep,name=globalParamRef"`
+
+	// NamespaceParamRef specifies the parameter resource or resources used to
+	// configure the admission control policy. It should point to a resources of
+	// the type specified in ParamKind of the bound ValidatingAdmissionPolicy.
+	//
+	// If the policy specifies a ParamKind and the resource referred to by
+	// ParamRef does not exist, this binding is considered mis-configured and
+	// the FailurePolicy of the ValidatingAdmissionPolicy is applied.
+	//
+	// if the paramKind of the policy referred to by `policyName` is cluster scoped,
+	// and namespaceParamRef set, the binding is considered mis-configured, and
+	// the failureMode applies.
+	//
+	// `namespaceParamRef` and `paramRef` are members of a union; if one of the
+	// fields is set, the other must be unset.
+	// +optional
+	NamespaceParamRef *NamespaceParamRef `json:"namespaceParamRef,omitempty" protobuf:"bytes,3,opt,name=namespaceParamRef"`
+}
+
+// BindingParamKind is the kind of ValidatingAdmissionPolicyBinding.
+// +enum
+type BindingParamKind string
+
+// Supported binding param kinds.
+const (
+	BindingParamKindGlobal    BindingParamKind = "Global"
+	BindingParamKindNamespace BindingParamKind = "Namespace"
+)
+
+// ParamRef references a parameter resource
+// +structType=atomic
+type ParamRef struct {
+	// Name of the resource being referenced.
+	Name string `json:"name,omitempty" protobuf:"bytes,1,rep,name=name"`
+	// Namespace of the referenced resource.
+	// Should be empty for the cluster-scoped resources
+	// +optional
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,rep,name=namespace"`
+}
+
+// NamespaceParamRef references a parameter resource scoped to the same
+// namespace as the resource being evaluated by a policy.
+// +structType=atomic
+type NamespaceParamRef struct {
```

Comment:
can we actually AND the name and label selector?  Thinking ahead to a future with field selector for CRs, I can imagine adding field selectors and I think that ANDing the output of all selection criteria is the easiest to reason about.

Reply:
added documentation to type for this detail, and removed note that name and selector are a union